programs/clearing_house/src/math/amm_jit.rs

```diff
@@ -9,26 +11,98 @@ use solana_program::msg;
 pub fn calculate_jit_base_asset_amount(
     market: &PerpMarket,
     taker_base_asset_amount: u128,
+    auction_price: u128,
+    valid_oracle_price: Option<i128>,
+    taker_direction: PositionDirection,
 ) -> ClearingHouseResult<u128> {
-    // simple impl
-    // todo: dynamic on imbalance
-    let mut jit_base_asset_amount = standardize_base_asset_amount(
+    let mut max_jit_amount = taker_base_asset_amount;
+
+    // check for wash trade
+    // simple impl: half max jit amount when likely to be wash
+    if let Some(oracle_price) = valid_oracle_price {
+        let oracle_price = cast_to_u128(oracle_price)?;
+
+        // maker taking a short below oracle = likely to be a wash
+        if taker_direction == PositionDirection::Long && auction_price < oracle_price {
+            max_jit_amount = max_jit_amount.checked_div(4).ok_or_else(math_error!())?
+        } else if taker_direction == PositionDirection::Short && auction_price > oracle_price {
+            max_jit_amount = max_jit_amount.checked_div(4).ok_or_else(math_error!())?
+        }
+    } else {
+        max_jit_amount = 0;
+    };
+
+    if max_jit_amount == 0 {
+        return Ok(0);
+    }
+
+    // check for market imbalance
+    // min/max_baa
+    let base_reserve_length = market
+        .amm
+        .max_base_asset_reserve
+        .checked_sub(market.amm.min_base_asset_reserve)
+        .ok_or_else(math_error!())?;
+
+    let half_base_reserve_length = base_reserve_length
+        .checked_div(2)
+        .ok_or_else(math_error!())?;
+
+    let mid_reserve = market
```

Comment:
this is the same as `terminal_base_asset_reserve`? would calc-ing the "bid/ask" liquidity of the amm be more clear whats happening here?